### PR TITLE
Comparing AFL++ variants

### PR DESCRIPTION
**Pull request by tokatoka**
Hi fuzzbench team!

This is another follow-up experiment for #1840 
in #1840, we compared LibAFL variants
and this time, we want to see how each technique performs on AFL++.

thanks.

### CODE REVIEW

**Comment by tokatoka** [comment]
Hello, @Alan32Liu 
This PR is ready. Could you run the experiment?

**Comment by Alan32Liu** [comment]
> Hello, @Alan32Liu This PR is ready. Could you run the experiment?

Of course!

**Comment by Alan32Liu** [comment]
/gcbrun run_experiment.py -a --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name 2023-07-20-aflpp-var --fuzzers aflplusplus_fuzzbench_cmplog aflplusplus_fuzzbench_ctx aflplusplus_fuzzbench_explore aflplusplus_fuzzbench_fast aflplusplus_fuzzbench_ngram4 aflplusplus_fuzzbench_ngram8 aflplusplus_fuzzbench_weighted

**Comment by Alan32Liu** [comment]
Would you mind making a trivial modification to [service/gcbrun_experiment.py](https://github.com/google/fuzzbench/blob/master/service/gcbrun_experiment.py)?
This will allow me to launch experiments in this PR before merging. Here is [an example to add a dummy comment](https://github.com/google/fuzzbench/pull/1818/files#diff-e870d8214beac56eaaeda2edc69de842f8382c428329dd8cf826a9cfa5edf7e9).

**Comment by tokatoka** [comment]
> Would you mind making a trivial modification to [service/gcbrun_experiment.py](https://github.com/google/fuzzbench/blob/master/service/gcbrun_experiment.py)?
Done

**Comment by Alan32Liu** [comment]
/gcbrun run_experiment.py -a --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name 2023-07-21-aflpp-var --fuzzers aflplusplus_fuzzbench_cmplog aflplusplus_fuzzbench_ctx aflplusplus_fuzzbench_explore aflplusplus_fuzzbench_fast aflplusplus_fuzzbench_ngram4 aflplusplus_fuzzbench_ngram8 aflplusplus_fuzzbench_weighted

**Comment by Alan32Liu** [comment]
Experiment data and results will be available at:
[The experiment data](https://storage.googleapis.com/fuzzbench-data/index.html?prefix=2023-07-21-aflpp-var/).
[The experiment report](https://storage.googleapis.com/www.fuzzbench.com/reports/experimental/2023-07-21-aflpp-var/index.html).

**Comment by tokatoka** [comment]
hmm
but this url looks empty
https://storage.googleapis.com/www.fuzzbench.com/reports/experimental/2023-07-21-aflpp-var/index.html

**Comment by Alan32Liu** [comment]
> hmm but this url looks empty https://storage.googleapis.com/www.fuzzbench.com/reports/experimental/2023-07-21-aflpp-var/index.html

This is likely due to the the [build failures with `aflplusplus_fuzzbench_cmplog`](https://storage.googleapis.com/fuzzbench-data/index.html?prefix=2023-07-21-aflpp-var/build-logs/).

The error seems to flaky, I will try relaunch another one.
Meanwhile, could you please also double-check if you could trigger it locally?
Thanks!

**Comment by Alan32Liu** [comment]
/gcbrun run_experiment.py -a --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name 2023-07-24-aflpp-var --fuzzers aflplusplus_fuzzbench_cmplog aflplusplus_fuzzbench_ctx aflplusplus_fuzzbench_explore aflplusplus_fuzzbench_fast aflplusplus_fuzzbench_ngram4 aflplusplus_fuzzbench_ngram8 aflplusplus_fuzzbench_weighted

**Comment by Alan32Liu** [comment]
Experiment data and results will be available at:
[The experiment data](https://storage.googleapis.com/fuzzbench-data/index.html?prefix=2023-07-24-aflpp-var/).
[The experiment report](https://storage.googleapis.com/www.fuzzbench.com/reports/experimental/2023-07-24-aflpp-var/index.html).

**Comment by tokatoka** [comment]
I cannot reproduce the error on my local computer. strange..

it seems like it is a network error; apt failing to fetch the deps... :man_shrugging: 
maybe it'll get fixed in a few days

**Comment by tokatoka** [comment]
@Alan32Liu  The experiment report is still empty.
And on my local computer I can't reproduce it (the network error is fixed now?)

Could you run it again?

**Comment by Alan32Liu** [comment]
> @Alan32Liu The experiment report is still empty. And on my local computer I can't reproduce the error shown in experiment data directory (the network error is fixed now?)
> 
> Could you run it again?

It's still having network error:
```
#5 - "aflplusplus_fuzzbench_cmplog-bloaty_fuzz_target-intermediate-runner": #5 4.353 Err:36 http://security.ubuntu.com/ubuntu focal-updates/main amd64 libcurl3-gnutls amd64 7.68.0-1ubuntu2.18
Step #5 - "aflplusplus_fuzzbench_cmplog-bloaty_fuzz_target-intermediate-runner": #5 4.353   404  Not Found [IP: 91.189.91.39 80]
Step #5 - "aflplusplus_fuzzbench_cmplog-bloaty_fuzz_target-intermediate-runner": #5 4.415 Fetched 47.6 MB in 2s (22.3 MB/s)
```

I found it in the `build-logs` dir in [the experiment data](https://storage.googleapis.com/fuzzbench-data/index.html?prefix=2023-07-24-aflpp-var/).

**Comment by Alan32Liu** [comment]
/gcbrun run_experiment.py -a --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name 2023-07-26-aflpp-var --fuzzers aflplusplus_fuzzbench_cmplog aflplusplus_fuzzbench_ctx aflplusplus_fuzzbench_explore aflplusplus_fuzzbench_fast aflplusplus_fuzzbench_ngram4 aflplusplus_fuzzbench_ngram8 aflplusplus_fuzzbench_weighted

**Comment by Alan32Liu** [comment]
Experiment data and results will be available at:
[The experiment data](https://storage.googleapis.com/fuzzbench-data/index.html?prefix=2023-07-26-aflpp-var/).
[The experiment report](https://storage.googleapis.com/www.fuzzbench.com/reports/experimental/2023-07-26-aflpp-var/index.html).

**Comment by Alan32Liu** [comment]
This is strange: The build steps failed again due to network problems.

File: fuzzers/aflplusplus_fuzzbench_cmplog/runner.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+FROM gcr.io/fuzzbench/base-image
+
+# This makes interactive docker runs painless:
+ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/out"
+#ENV AFL_MAP_SIZE=2621440
+ENV PATH="$PATH:/out"
+ENV AFL_SKIP_CPUFREQ=1
+ENV AFL_I_DONT_CARE_ABOUT_MISSING_CRASHES=1
+ENV AFL_TESTCACHE_SIZE=2
+RUN apt install -y unzip git gdb joe
```
Comment (Alan32Liu) [inline fuzzers/aflplusplus_fuzzbench_cmplog/runner.Dockerfile:24]: According to the error log, this is the reason for the build failure.
Could you please add an `apt-get update` statement before this? (i.e., `apt-get update; apt-get install unzip git gdb joe`)
It's also better to replace apt with apt-get.
Same for other fuzzers.

That solves the issue when I test it in the docker container.

**Review by Alan32Liu — APPROVED** [review]

**Comment by Alan32Liu** [comment]
/gcbrun run_experiment.py -a --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name 2023-07-27-aflpp-var --fuzzers aflplusplus_fuzzbench_cmplog aflplusplus_fuzzbench_ctx aflplusplus_fuzzbench_explore aflplusplus_fuzzbench_fast aflplusplus_fuzzbench_ngram4 aflplusplus_fuzzbench_ngram8 aflplusplus_fuzzbench_weighted

**Comment by Alan32Liu** [comment]
Experiment data and results will be available at:
[The experiment data](https://storage.googleapis.com/fuzzbench-data/index.html?prefix=2023-07-27-aflpp-var/).
[The experiment report](https://storage.googleapis.com/www.fuzzbench.com/reports/experimental/2023-07-27-aflpp-var/index.html).

**Comment by tokatoka** [comment]
Thank you!

**Pull request closed without merging**